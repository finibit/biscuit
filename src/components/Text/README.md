### Usage

#### As

```jsx
<Text as="div">Text</Text>
```

#### Color

```jsx
<Text color="info">Text</Text>
```
```jsx
<Text color="#009900">Text</Text>
```
```jsx
<Text color="violet">Text</Text>
```

#### Weight

```jsx
<Text weight="bold">Text</Text>
```
```jsx
<Text weight={700}>Text</Text>
```

#### Align

```jsx
<Text as="div" align="left">Text</Text>
```
```jsx
<Text as="div" align="right">Text</Text>
```
```jsx
<Text as="div" align="center">Text</Text>
```

#### Italic

```jsx
<Text italic>Text</Text>
```

#### Transform

```jsx
<Text transform="uppercase">Text</Text>
```

```jsx
<Text transform="lowercase">HEADING</Text>
```

```jsx
<Text transform="capitalize">heading</Text>
```

```jsx
<Text transform="full-width">Text</Text>
```

#### Decoration

```jsx
<Text decoration="underline">Text</Text>
```

```jsx
<Text decoration="overline">Text</Text>
```

```jsx
<Text decoration="line-through">Text</Text>
```

#### No wrap

```jsx
<Text as="div" nowrap>Bits of moving fluff tesseract vastness is bearable only through love quasar rich in heavy atoms Euclid. The only home we've ever known realm of the galaxies ship of the imagination realm of the galaxies the carbon in our apple pies finite but unbounded? Encyclopaedia galactica are creatures of the cosmos across the centuries citizens of distant epochs citizens of distant epochs a mote of dust suspended in a sunbeam and billions upon billions upon billions upon billions upon billions upon billions upon billions.</Text>
```

#### Size

```jsx
<Text size={4}>Text</Text>
```
```jsx
<Text size="2rem">Text</Text>
```
```jsx
<Text size={[0, 1, 2, 3]}>Text</Text>
```

#### Margin

```jsx
<Text as="div" margin="50px">Text</Text>
```

```jsx
<Text as="div" margin={3}>Text</Text>
```

```jsx
<Text as="div" margin={[0, 1, 2, 3]}>Text</Text>
```

#### Padding

```jsx
<Text as="div" padding={3}>Text</Text>
```

#### Css

```jsx
<Text css="border 1px solid #000;">Text</Text>
```

```jsx
<Text css={{ border: "1px solid #000;" }}>Text</Text>
```

```jsx
import { css } from 'styled-components';

const style = css`
  color: ${(props) => props.theme.global.colors.info};
  border: 1px solid ${(props) => props.theme.global.colors.info};
`;

<Text css={style}>Text</Text>
```
