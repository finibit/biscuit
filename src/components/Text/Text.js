import React from 'react'
import PropTypes from 'prop-types'
import styled, { css as styledCss } from 'styled-components'
import styles from '../../themes'

const nowrapStyles = styledCss`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`

const TextStyled = styled.span`
  ${styles.fontFamily}
  ${styles.fontWeight}
  ${styles.fontStyle}
  ${styles.fontSize}
  ${styles.lineHeight}
  ${styles.textAlign}
  ${styles.textDecoration}
  ${styles.textTransform}
  ${styles.spacing}
  ${styles.color}
  ${(props) => props.$nowrap && nowrapStyles}
  ${(props) => props.$css}
`

/** A non-semantic text. */
const Text = (props) => {
  const {
    color,
    align,
    nowrap,
    weight,
    italic,
    transform,
    decoration,
    size,
    margin,
    padding,
    themeElement,
    css,
    ...rest
  } = props

  return (
    <TextStyled
      $element={themeElement}
      $color={color}
      $align={align}
      $nowrap={nowrap}
      $weight={weight}
      $italic={italic}
      $transform={transform}
      $decoration={decoration}
      $size={size}
      $margin={margin}
      $padding={padding}
      $css={css}
      {...rest}
    />
  )
}

Text.propTypes = {
  /** Any number of renderable nodes. */
  children: PropTypes.node,

  /** DOM element to use. */
  as: PropTypes.string,

  /** Text color. */
  color: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.shape({
      value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      shade: PropTypes.number,
      tint: PropTypes.number
    })
  ]),

  /** Text alignment. */
  align: PropTypes.oneOf(['left', 'center', 'right']),

  /** Should the text wrap if it doesn't fit into container. */
  nowrap: PropTypes.bool,

  /** Font weight. */
  weight: PropTypes.oneOfType([PropTypes.oneOf(['normal', 'bold']), PropTypes.number]),

  /** Use italic version of the font. */
  italic: PropTypes.bool,

  /** Text transformation. */
  transform: PropTypes.oneOf(['none', 'uppercase', 'lowercase', 'capitalize', 'full-width']),

  /** Text decoration. */
  decoration: PropTypes.oneOf(['none', 'underline', 'overline', 'line-through']),

  /** Size of the text. */
  size: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /** The amount of margin around the text. */
  margin: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /** The amount of padding around the text. */
  padding: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /** Theme element. */
  themeElement: PropTypes.string,

  /** Custom styles passed to styled-components. */
  css: PropTypes.oneOfType([PropTypes.string, PropTypes.object, PropTypes.array])
}

Text.defaultProps = {
  children: null,
  as: 'span',
  color: 0,
  weight: 'normal',
  align: 'left',
  italic: false,
  transform: 'none',
  decoration: 'none',
  nowrap: false,
  size: 2,
  margin: 'none',
  padding: 'none',
  themeElement: 'Text',
  css: null
}

export default Text
