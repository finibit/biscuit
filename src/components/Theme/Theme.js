import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider, createGlobalStyle } from 'styled-components'
import { mergeTheme, baseTheme, themeGet } from '../../themes'
import types from '../../types'

const GlobalStyle = createGlobalStyle`
	html, body {
		width: 100%;
		height: 100%;
		margin: 0;
		padding: 0;
	}

	body {
		color: ${themeGet.color('black', 'inherit')};
		font-family: ${themeGet.fontFamily('secondary')};
		-webkit-font-smoothing: antialiased;
	}
`

const Theme = (props) => {
	const {
		theme,
		global,
	} = props

	return (
		<ThemeProvider theme={mergeTheme(baseTheme, theme)}>
			<div>
				{ global && <GlobalStyle /> }
				{props.children}
			</div>
		</ThemeProvider>
	)
}

Theme.propTypes = {
	global: PropTypes.bool,
	...types.theme,
}

Theme.defaultProps = {
	global: true,
	theme: baseTheme,
}

export default Theme
