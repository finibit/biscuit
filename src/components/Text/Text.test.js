import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Theme from '../Theme'
import Text from '../Text'

describe('Text', () => {
	test('default', () => {
		const component = renderer.create((
			<Theme>
				<Text />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('as', () => {
		const component = renderer.create((
			<Theme>
				<Text as="div" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('color', () => {
		const component = renderer.create((
			<Theme>
				<Text color="light-0" />
				<Text color="#ffaa00" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('align', () => {
		const component = renderer.create((
			<Theme>
				<Text align="left" />
				<Text align="right" />
				<Text align="center" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('nowrap', () => {
		const component = renderer.create((
			<Theme>
				<Text nowrap />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('weight', () => {
		const component = renderer.create((
			<Theme>
				<Text weight="normal" />
				<Text weight="bold" />
				<Text weight={600} />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('italic', () => {
		const component = renderer.create((
			<Theme>
				<Text italic />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('transform', () => {
		const component = renderer.create((
			<Theme>
				<Text transform="none" />
				<Text transform="uppercase" />
				<Text transform="lowercase" />
				<Text transform="capitalize" />
				<Text transform="full-width" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('decoration', () => {
		const component = renderer.create((
			<Theme>
				<Text decoration="none" />
				<Text decoration="underline" />
				<Text decoration="overline" />
				<Text decoration="line-through" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('size', () => {
		const component = renderer.create((
			<Theme>
				<Text size="xs" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('size responsive', () => {
		const component = renderer.create((
			<Theme>
				<Text size={['xs', 'sm', 'md', 'lg']} />
				<Text size={{ xs: 'xs', sm: 'sm', md: 'md', lg: 'lg' }} />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('padding', () => {
		const component = renderer.create((
			<Theme>
				<Text padding="xs" />
				<Text padding={{}} />
				<Text padding={{ top: 'xs' }} />
				<Text padding={{ right: 'xs' }} />
				<Text padding={{ bottom: 'xs' }} />
				<Text padding={{ left: 'xs' }} />
				<Text padding={{ horizontal: 'xs' }} />
				<Text padding={{ vertical: 'xs' }} />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('padding responsive', () => {
		const component = renderer.create((
			<Theme>
				<Text padding={['xs', 'sm', 'md', 'lg']} />
				<Text padding={{ xs: 'xs', sm: 'sm', md: 'md', lg: 'lg' }} />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('margin', () => {
		const component = renderer.create((
			<Theme>
				<Text margin="xs" />
				<Text margin={{}} />
				<Text margin={{ top: 'xs' }} />
				<Text margin={{ right: 'xs' }} />
				<Text margin={{ bottom: 'xs' }} />
				<Text margin={{ left: 'xs' }} />
				<Text margin={{ horizontal: 'xs' }} />
				<Text margin={{ vertical: 'xs' }} />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('margin responsive', () => {
		const component = renderer.create((
			<Theme>
				<Text margin={['xs', 'sm', 'md', 'lg']} />
				<Text margin={{ xs: 'xs', sm: 'sm', md: 'md', lg: 'lg' }} />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('css', () => {
		const component = renderer.create((
			<Theme>
				<Text css="background-color: #000;" />
				<Text css={{ backgroundColor: '#000' }} />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})
})
