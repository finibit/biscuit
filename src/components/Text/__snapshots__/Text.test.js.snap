// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text align 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: right;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

<div>
  <span
    className="c0"
  />
  <span
    className="c1"
  />
  <span
    className="c2"
  />
</div>
`;

exports[`Text as 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

<div>
  <div
    className="c0"
  />
</div>
`;

exports[`Text color 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: light-0;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #ffaa00;
}

<div>
  <span
    className="c0"
  />
  <span
    className="c1"
  />
</div>
`;

exports[`Text css 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
  background-color: #000;
}

<div>
  <span
    className="c0"
  />
  <span
    className="c0"
  />
</div>
`;

exports[`Text decoration 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: overline;
  text-decoration: overline;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c3 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

<div>
  <span
    className="c0"
  />
  <span
    className="c1"
  />
  <span
    className="c2"
  />
  <span
    className="c3"
  />
</div>
`;

exports[`Text default 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

<div>
  <span
    className="c0"
  />
</div>
`;

exports[`Text italic 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: italic;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

<div>
  <span
    className="c0"
  />
</div>
`;

exports[`Text margin 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: xs;
  padding: 0;
  color: #000000;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: initial;
  padding: 0;
  color: #000000;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: xsrem 0 0 0;
  padding: 0;
  color: #000000;
}

.c3 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0 xsrem 0 0;
  padding: 0;
  color: #000000;
}

.c4 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0 0 xsrem 0;
  padding: 0;
  color: #000000;
}

.c5 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0 0 0 xsrem;
  padding: 0;
  color: #000000;
}

.c6 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0 xsrem 0 xsrem;
  padding: 0;
  color: #000000;
}

.c7 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: xsrem 0 xsrem 0;
  padding: 0;
  color: #000000;
}

<div>
  <span
    className="c0"
  />
  <span
    className="c1"
  />
  <span
    className="c2"
  />
  <span
    className="c3"
  />
  <span
    className="c4"
  />
  <span
    className="c5"
  />
  <span
    className="c6"
  />
  <span
    className="c7"
  />
</div>
`;

exports[`Text margin responsive 1`] = `
.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: initial;
  padding: 0;
  color: #000000;
}

.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: xs;
  padding: 0;
  color: #000000;
}

@media only screen and (min-width:48rem) {
  .c0 {
    margin: sm;
  }
}

@media only screen and (min-width:64rem) {
  .c0 {
    margin: md;
  }
}

@media only screen and (min-width:75rem) {
  .c0 {
    margin: lg;
  }
}

<div>
  <span
    className="c0"
  />
  <span
    className="c1"
  />
</div>
`;

exports[`Text nowrap 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div>
  <span
    className="c0"
  />
</div>
`;

exports[`Text padding 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: xs;
  color: #000000;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: initial;
  color: #000000;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: xsrem 0 0 0;
  color: #000000;
}

.c3 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0 xsrem 0 0;
  color: #000000;
}

.c4 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0 0 xsrem 0;
  color: #000000;
}

.c5 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0 0 0 xsrem;
  color: #000000;
}

.c6 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0 xsrem 0 xsrem;
  color: #000000;
}

.c7 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: xsrem 0 xsrem 0;
  color: #000000;
}

<div>
  <span
    className="c0"
  />
  <span
    className="c1"
  />
  <span
    className="c2"
  />
  <span
    className="c3"
  />
  <span
    className="c4"
  />
  <span
    className="c5"
  />
  <span
    className="c6"
  />
  <span
    className="c7"
  />
</div>
`;

exports[`Text padding responsive 1`] = `
.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: initial;
  color: #000000;
}

.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: xs;
  color: #000000;
}

@media only screen and (min-width:48rem) {
  .c0 {
    padding: sm;
  }
}

@media only screen and (min-width:64rem) {
  .c0 {
    padding: md;
  }
}

@media only screen and (min-width:75rem) {
  .c0 {
    padding: lg;
  }
}

<div>
  <span
    className="c0"
  />
  <span
    className="c1"
  />
</div>
`;

exports[`Text size 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: xs;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

<div>
  <span
    className="c0"
  />
</div>
`;

exports[`Text size responsive 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: xs;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: initial;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

@media only screen and (min-width:48rem) {
  .c0 {
    font-size: sm;
  }
}

@media only screen and (min-width:64rem) {
  .c0 {
    font-size: md;
  }
}

@media only screen and (min-width:75rem) {
  .c0 {
    font-size: lg;
  }
}

<div>
  <span
    className="c0"
  />
  <span
    className="c1"
  />
</div>
`;

exports[`Text transform 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: lowercase;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c3 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c4 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: full-width;
  margin: 0;
  padding: 0;
  color: #000000;
}

<div>
  <span
    className="c0"
  />
  <span
    className="c1"
  />
  <span
    className="c2"
  />
  <span
    className="c3"
  />
  <span
    className="c4"
  />
</div>
`;

exports[`Text weight 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: normal;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: bold;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-style: normal;
  font-size: 1rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  margin: 0;
  padding: 0;
  color: #000000;
}

<div>
  <span
    className="c0"
  />
  <span
    className="c1"
  />
  <span
    className="c2"
  />
</div>
`;
